src/librustc/mir/visit.rs

```diff
@@ -725,15 +725,18 @@ macro_rules! make_mir_visitor {
                             place: & $($mutability)? Place<'tcx>,
                             context: PlaceContext<'tcx>,
                             location: Location) {
+                use crate::mir::{Static, StaticKind};
                 match place {
                     Place::Base(PlaceBase::Local(local)) => {
                         self.visit_local(local, context, location);
                     }
-                    Place::Base(PlaceBase::Static(static_)) => {
-                        if static_.promoted.is_none() {
-                            self.visit_def_id(& $($mutability)? static_.def_id, location);
-                        }
-                        self.visit_ty(& $($mutability)? static_.ty, TyContext::Location(location));
+                    Place::Base(
+                        PlaceBase::Static(box Static{kind: StaticKind::Static(def_id), ..})
+                    ) => {
+                        self.visit_def_id(& $($mutability)? *def_id, location)
```

Comment:
you need to visit the type here, too. Instead of doing that twice, you can just use the code as you had it before this PR and do `if let StaticKind::Static(def_id) = static_.kind {`